src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/NoHardcodedLocationRuleTests.cs

```diff
@@ -0,0 +1,591 @@
+﻿// Copyright (c) Microsoft Corporation.
```

Comment:
Bug:

```bicep
module m1 'module1.bicep' = {
  name: 'module1'
  params: {
    location: 'westus'    
  }
}
```
module1.bicep:
```bicep
param location string

resource storageaccount 'Microsoft.Storage/storageAccounts@2021-02-01' = {
  name: 'storageaccount'
  location: location
  kind: 'StorageV2'
  sku: {
    name: 'Premium_LRS'
  }
}
```
Expected: no-hardcoded-location should fail, but doesn't